README.md

```diff
@@ -230,6 +230,43 @@ jobs:
 ```
 </details>
 
+## Deleting caches
+We can not re-use caches from pull request branches in other branches like main, such caches can eat up the storage quota and hence causing thrashing on more useful branches like main. In order to resolve this issue, we can use [gh-actions-cache cli](https://github.com/actions/gh-actions-cache/) to delete caches. This workflow uses `gh-actions-cache` to delete all the caches created by all the pull requests. 
+
+```
+name: cleanup-caches
+on:
+  workflow_dispatch
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+
+      - name: Cleanup
+        run: |
+          gh extension install actions/gh-actions-cache
+          
+          REPO=${{ github.repository }}
+
+          echo "Fetching list of cache key"
+          ## This will extract out all the cache keys for pull requests
+          cacheKeysForPR=$(gh actions-cache list -R $REPO | grep "refs/pull" | cut -d $'\t'  -f 1 )
```

Comment:
Should we only unique cache keys so that we do not try to delete same key again.

Reply:
Just sticking with -B input rather than grep